### PR TITLE
Add Beyond Identity Provider

### PR DESCRIPTION
**Pull request by Will5**
## ☕️ Reasoning

This pull request adds a BeyondIdentityProvider and the corresponding documentation which allows authenticating with [Beyond Identity](https://www.beyondidentity.com/).

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

**Comment by Will5** [comment]
This originally only contained an "oauth" provider, but I added an "oidc" provider as well. This required a fix for the next-auth.pkce.code_verifier cookie, which should also address Issue #5101 as well.

**Comment by Will5** [comment]
I merged in the latest master, resolved the latest merge conflicts, and retested via the example app. Is there anything else required? Thanks!

File: packages/core/src/lib/oauth/pkce-handler.ts
Comment (ThangHuuVu) [inline packages/core/src/lib/oauth/pkce-handler.ts:83]: This should not be changed, did you run into an issue? if so, please open a separate PR for this

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]
Hi there @Will5 thanks for sending the PR! To keep it up to date and ready to merge, here are a few things that need to be done:
- Revert all changes in `packages/next-auth` and move them to `packages/core`
- Don't use `.md` doc, instead use JSDoc like https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/auth0.ts
- Add the provider to the drop down at https://github.com/nextauthjs/next-auth/blob/main/.github/ISSUE_TEMPLATE/2_bug_provider.yml
LMK if you need any help 🙌

File: packages/core/src/lib/oauth/pkce-handler.ts
Comment (Will5) [reply packages/core/src/lib/oauth/pkce-handler.ts:83]: Yes, I don't remember the details, but PKCE was not working correctly without it. If I recall correctly, the code was empty in the token exchange, which seemed to be because the cookie was empty. However, I tried just now with the latest, and I did not encounter an issue. I will revert, and open an issue if I come across it again.

Comment (Will5) [reply packages/core/src/lib/oauth/pkce-handler.ts:83]: Oh, I think I encountered the issue again. The pkce challenge was not validated. It seems to be related to checking both "pkce" and "state". I did not encounter the issue when I just checked "pkce". That being said, I don't think this is the most appropriate solution.

**Comment by Will5** [comment]
Hi @ThangHuuVu, just checking in. Is there anything else needed from me? Would love to get this in. Thanks!

**Comment by ThangHuuVu** [comment]
@Will5 sorry for the late response, could you update the branch before I merge it? All LGTM 🙌

**Comment by Will5** [comment]
Hi @ThangHuuVu, done. Let me know if you need anything else. Thanks 🙌

**Review by ThangHuuVu — APPROVED** [review]

File: packages/core/src/providers/beyondidentity.ts
Comment (balazsorban44) [inline packages/core/src/providers/beyondidentity.ts:84]:
```suggestion
        image: null,
```

File: packages/core/src/providers/beyondidentity.ts
Comment (balazsorban44) [inline packages/core/src/providers/beyondidentity.ts:88]:
```suggestion
```

**Comment by ThangHuuVu** [comment]
Can you attach a video for proof? @Will5 🙌

**Comment by Will5** [comment]
@ThangHuuVu, below is a video using the dev app, though it seems the header remains in signed out state. This is the same when I use Auth0 as well, so I suspect it's a dev app bug. Should I submit a bug for this?

https://user-images.githubusercontent.com/4763133/220440859-7fe884ff-dc83-496f-b40d-17db5fa3ca8d.mov

**Comment by ThangHuuVu** [comment]
@Will5 I see, we are actually not sure if it's a bug - it only happens on the local development. If you un-comment this line, it should work: https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/web.ts#L81

**Comment by Will5** [comment]
@ThangHuuVu Thanks for the tip! That worked for me 🙂 


https://user-images.githubusercontent.com/4763133/220812702-16b54628-c148-43f7-a8bd-320bacba479f.mov

**Comment by Will5** [comment]
Hi @ThangHuuVu Thanks for your help! I notice the provider isn't showing up in the docs. Does it need to wait for the next release? If so, is there an expected time frame? Thanks!

**Comment by Will5** [comment]
Never mind, I see it now! 🎉

**Pull request merged**
